cli/pkg/validate/entity/entity_validator.go

```diff
@@ -0,0 +1,67 @@
+package entity
+
+import (
+	"errors"
+
+	"github.com/rudderlabs/rudder-iac/cli/pkg/localcatalog"
+	"github.com/rudderlabs/rudder-iac/cli/pkg/logger"
+)
+
+var (
+	log = logger.New("entity.validate")
+)
+
+var (
+	// Common Errors
+	ErrDuplicateByID           = errors.New("duplicate entity by id")
+	ErrDuplicateByName         = errors.New("duplicate entity by display_name")
+	ErrDuplicateByNameType     = errors.New("duplicate entity by name and type")
+	ErrMissingRequiredKeysID   = errors.New("missing required key id")
+	ErrMissingRequiredKeysName = errors.New("missing required key display_name")
```

Comment:
Some of these errors look generic (e.g duplicate entity by id) while others look specific to this type of entities (e.g display_name needs to be unique). We should clearly separate those two and allow specific entities to have their own specific validation rules if necessary.